From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Fri, 25 May 2018 09:13:44 +0200
Subject: [PATCH] egl: autotools: add missing dependency on generated header

platform_wayland.c includes linux-dmabuf-unstable-v1-client-protocol.h,
which is generated during build. Add the missing dependency to the
Makefile.

I have seen the following build failure due to a race between generation
of linux-dmabuf-unstable-v1-client-protocol.h and compilation of
platform_wayland.cc:

  GEN      drivers/dri2/linux-dmabuf-unstable-v1-client-protocol.h
  GEN      drivers/dri2/linux-dmabuf-unstable-v1-protocol.c
Using "code" is deprecated - use private-code or public-code.
See the help page for details.
  CC       drivers/dri2/platform_wayland.lo
../../../Mesa-18.1.0/src/egl/drivers/dri2/platform_wayland.c: In function 'create_wl_buffer':
../../../Mesa-18.1.0/src/egl/drivers/dri2/platform_wayland.c:810:16: error: implicit declaration of function 'zwp_linux_dmabuf_v1_create_params' [-Werror=implicit-function-declaration]

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/egl/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/egl/Makefile.am b/src/egl/Makefile.am
index 086a4a1e6302..116ed4ebf50a 100644
--- a/src/egl/Makefile.am
+++ b/src/egl/Makefile.am
@@ -80,6 +80,7 @@ drivers/dri2/linux-dmabuf-unstable-v1-client-protocol.h: $(WL_DMABUF_XML)
 if HAVE_PLATFORM_WAYLAND
 drivers/dri2/linux-dmabuf-unstable-v1-protocol.lo: drivers/dri2/linux-dmabuf-unstable-v1-client-protocol.h
 drivers/dri2/egl_dri2.lo: drivers/dri2/linux-dmabuf-unstable-v1-client-protocol.h
+drivers/dri2/platform_wayland.lo: drivers/dri2/linux-dmabuf-unstable-v1-client-protocol.h
 
 AM_CFLAGS += $(WAYLAND_CLIENT_CFLAGS)
 libEGL_common_la_LIBADD += $(WAYLAND_CLIENT_LIBS)
